### ISSUE TITLE
"identity stage" not correctly reproduced when writing and reading Inventory

### ISSUE THREAD

**Issue by trichter** [obspy/obspy#2250]
Obspy master does not write the output and input units of identity stages, i.e. stages that do nothing. As a consequence, input and output units are set to None when reading the inventory again.

```py
from obspy import read_inventory, UTCDateTime as UTC
from obspy.clients.nrl import NRL
from obspy.core.inventory import Inventory, Network, Station, Channel, Site

nrl = NRL()
sensor = ['Guralp', 'CMG-3ESP', '60 s - 50 Hz', '2000']
datalogger = ['REF TEK', 'RT 130 & 130-SMA', '1', '100']
response = nrl.get_response(datalogger, sensor)
print('original reponse')
print(response)

ch = Channel(code='HHZ', location_code='', latitude=0, longitude=0,
             elevation=0, depth=0, sample_rate=100, response=response)
st = Station(code='XX', latitude=0, longitude=0, elevation=0,
             site=Site('XX'), channels=[ch])
net = Network(code='YY', stations=[st])
inv = Inventory(networks=[net], source='me')
print('\nstill OK after creating Inventory')
print(inv.get_response('YY.XX..HHZ', UTC()))

inv.write('bug.xml', 'STATIONXML')
inv2 = read_inventory('bug.xml')
response2 = inv2.get_response('YY.XX..HHZ', UTC())
print('\nunits missing in stage 2 after writing and reading')
print(response2)
```

Output:
```
original reponse
Channel Response
	From M/S (Velocity in Meters per Second) to COUNTS (Digital Counts)
	Overall Sensitivity: 1.25932e+09 defined at 1.000 Hz
	11 stages:
		Stage 1: PolesZerosResponseStage from M/S to V, gain: 2000
		Stage 2: ResponseStage from V to V, gain: 1
		Stage 3: CoefficientsTypeResponseStage from V to COUNTS, gain: 629129
		Stage 4: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 5: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 6: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 7: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 8: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 9: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 10: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 11: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1

still OK after creating Inventory
Channel Response
	From M/S (Velocity in Meters per Second) to COUNTS (Digital Counts)
	Overall Sensitivity: 1.25932e+09 defined at 1.000 Hz
	11 stages:
		Stage 1: PolesZerosResponseStage from M/S to V, gain: 2000
		Stage 2: ResponseStage from V to V, gain: 1
		Stage 3: CoefficientsTypeResponseStage from V to COUNTS, gain: 629129
		Stage 4: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 5: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 6: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 7: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 8: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 9: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 10: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 11: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1

units missing in stage 2 after writing and reading
Channel Response
	From M/S (Velocity in Meters per Second) to COUNTS (Digital Counts)
	Overall Sensitivity: 1.25932e+09 defined at 1.000 Hz
	11 stages:
		Stage 1: PolesZerosResponseStage from M/S to V, gain: 2000
		Stage 2: ResponseStage from None to None, gain: 1
		Stage 3: CoefficientsTypeResponseStage from V to COUNTS, gain: 629129
		Stage 4: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 5: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 6: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 7: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 8: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 9: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 10: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
		Stage 11: CoefficientsTypeResponseStage from COUNTS to COUNTS, gain: 1
```

**Comment by trichter** [comment]
Not sure, if this should be fixed when writing or reading the XML. I guess the XML is OK and it should be fixed upon reading?

**Comment by trichter** [comment]
Here is a minimal example

```py
from obspy import UTCDateTime as UTC, read_inventory
from obspy.core.inventory.response import ResponseStage

utc = UTC('2017-01-01')
inv = read_inventory().select(station='RJOB', channel='EHZ', time=utc)
response = inv.get_response('BW.RJOB..EHZ', utc)
rstage2 = ResponseStage(2, 1, 1, 'V', 'V')
rstage3 = ResponseStage(3, 1, 1, 'V', 'V')
response.response_stages.insert(1, rstage2)
response.response_stages.insert(2, rstage3)
for i, rstage in enumerate(response.response_stages[3:]):
    rstage.stage_sequence_number = i + 4
inv.write('test.xml', 'STATIONXML')
inv2 = read_inventory('test.xml')
response2 = inv2.get_response('BW.RJOB..EHZ', utc)
print(response)
print(response2)
```

```
Channel Response
	From M/S (Velocity in Meters per Second) to COUNTS (Digital Counts)
	Overall Sensitivity: 2.5168e+09 defined at 0.020 Hz
	6 stages:
		Stage 1: PolesZerosResponseStage from M/S to V, gain: 1500
		Stage 2: ResponseStage from V to V, gain: 1
		Stage 3: ResponseStage from V to V, gain: 1
		Stage 4: CoefficientsTypeResponseStage from V to COUNTS, gain: 1.67785e+06
		Stage 5: FIRResponseStage from COUNTS to COUNTS, gain: 1
		Stage 6: FIRResponseStage from COUNTS to COUNTS, gain: 1
Channel Response
	From M/S (Velocity in Meters per Second) to COUNTS (Digital Counts)
	Overall Sensitivity: 2.5168e+09 defined at 0.020 Hz
	6 stages:
		Stage 1: PolesZerosResponseStage from M/S to V, gain: 1500
		Stage 2: ResponseStage from None to None, gain: 1
		Stage 3: ResponseStage from None to None, gain: 1
		Stage 4: CoefficientsTypeResponseStage from V to COUNTS, gain: 1.67785e+06
		Stage 5: FIRResponseStage from COUNTS to COUNTS, gain: 1
		Stage 6: FIRResponseStage from COUNTS to COUNTS, gain: 1
```

**Comment by megies** [comment]
> Not sure, if this should be fixed when writing or reading the XML. I guess the XML is OK and it should be fixed upon reading?

Looking at the `test.xml` you produce there, it misses the unit descriptions. So it needs to be fixed upon writing the StationXML.

**Comment by megies** [comment]
So I guess there's a mismatch in ObsPy's `ResponseStage` object (can always have I/O units) and StationXML's `ResponseStage` definition (gain-only stages won't have the possibility to hold I/O units).

**Comment by krischer** [comment]
Fixed in #2318.
